script/spec-runner.js

```diff
@@ -141,6 +143,70 @@ async function runRemoteBasedElectronTests () {
   console.log(`${pass} Electron remote process tests passed.`)
 }
 
+async function runNativeElectronTests () {
+  const TEST_TARGETS = require('./native-test-targets.json')
+  const outDir = `out/${utils.getOutDir(false)}`
+
+  // If native tests are being run, only one arg would be relevant
+  if (args.target && !TEST_TARGETS.includes(args.target)) {
+    console.log(`${fail} ${args.target} must be a subset of [${[TEST_TARGETS].join(', ')}]`)
+    process.exit(1)
+  }
+
+  // Optionally build all native test targets
+  if (args.buildNativeTests) {
+    for (const target of TEST_TARGETS) {
+      const build = childProcess.spawnSync('ninja', ['-C', outDir, target], {
+        cwd: path.resolve(__dirname, '../..'),
+        stdio: 'inherit'
+      })
+
+      // Exit if test target failed to build
+      if (build.status !== 0) {
+        console.log(`${fail} ${target} failed to build.`)
+        process.exit(1)
+      }
+    }
+  }
+
+  // If a specific target was passed, only build and run that target
+  if (args.target) {
+    console.info('\nRunning native test for target:', args.target)
+
+    // Run single test target
+    const run = childProcess.spawnSync(`./${outDir}/${args.target}`, {
+      cwd: path.resolve(__dirname, '../..'),
+      stdio: 'inherit'
+    })
+
+    if (run.status !== 0) {
+      console.log(`${fail} ${args.target} test failed.`)
+      process.exit(1)
+    }
+  } else {
```

Comment:
I think we can deduplicate this so that we share logic between the "one test" and "all tests" cases.